core/src/Kokkos_Concepts.hpp

```diff
@@ -286,9 +286,10 @@ struct is_space {
 
  public:
   static constexpr bool value = is_exe::value || is_mem::value || is_dev::value;
-
+  static constexpr bool is_exec_space() { return is_exe::value; }
+  static constexpr bool is_mem_space() { return is_mem::value; }
+  static constexpr bool is_device() { return is_dev::value; }
```

Comment:
Not sure how I feel about this given that this is supposed to be a minimal type trait, on the other hand it does contain to public typedefs already. Other folks thoughts?

Reply:
Also assume we keep this: are these tested?